payments-core/src/main/java/com/stripe/android/payments/core/authentication/DefaultIntentAuthenticatorRegistry.kt

```diff
@@ -0,0 +1,116 @@
+                // can only triggered when `use_stripe_sdk=true`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.RedirectToUrl -> {
+                // can only triggered when `use_stripe_sdk=false`
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.AlipayRedirect -> {
+                return webIntentAuthenticator
+            }
+            is StripeIntent.NextActionData.DisplayOxxoDetails -> {
+                return webIntentAuthenticator.takeIf { nextActionData.hostedVoucherUrl != null }
+                    ?: noOpIntentAuthenticator
+            }
+            else -> return noOpIntentAuthenticator
+        }
+    }
+
+    companion object {
+        // Holding this single instance would in turn holds DaggerAuthenticationComponent instance,
+        // which keeps dagger injection graph live.
+        private var instance: IntentAuthenticatorRegistry? = null
+
+        /**
+         * Return the singleton instance of [IntentAuthenticatorRegistry].
+         */
+        fun getInstance(
+            stripeRepository: StripeRepository,
+            paymentRelayStarterFactory: (AuthActivityStarter.Host) -> PaymentRelayStarter,
+            paymentBrowserAuthStarterFactory: (AuthActivityStarter.Host) -> PaymentBrowserAuthStarter,
+            analyticsRequestExecutor: AnalyticsRequestExecutor,
+            analyticsRequestFactory: AnalyticsRequestFactory,
+            logger: Logger,
+            enableLogging: Boolean,
+            workContext: CoroutineContext,
+            uiContext: CoroutineContext,
+            threeDs2Service: StripeThreeDs2Service,
+            messageVersionRegistry: MessageVersionRegistry,
+            challengeProgressActivityStarter: StripePaymentController.ChallengeProgressActivityStarter,
+            stripe3ds2Config: PaymentAuthConfig.Stripe3ds2Config,
+            stripe3ds2ChallengeLauncher: ActivityResultLauncher<PaymentFlowResult.Unvalidated>?
+        ): IntentAuthenticatorRegistry {
+            if (instance == null) {
+                instance = DaggerAuthenticationComponent.builder().authenticationModule(
+                    AuthenticationModule(
+                        stripeRepository,
+                        paymentRelayStarterFactory,
+                        paymentBrowserAuthStarterFactory,
+                        analyticsRequestExecutor,
+                        analyticsRequestFactory,
+                        logger,
+                        enableLogging,
+                        workContext,
+                        uiContext,
+                        threeDs2Service,
+                        messageVersionRegistry,
+                        challengeProgressActivityStarter,
+                        stripe3ds2Config,
+                        stripe3ds2ChallengeLauncher
+                    )
+                ).build().registry
+            }
+            return instance!!
```

Comment:
nit: can we do `fun getInstance(...) = instance ?: createInstance()` and move the object creation and caching into `createInstance()`